packages/docs-reanimated/docs/guides/contributing.mdx

```diff
@@ -0,0 +1,318 @@
+
+:::caution
+
+Make sure that you have [React Native setup](https://reactnative.dev/docs/environment-setup) properly and you got familiar with [How to Open a Pull Request](https://reactnative.dev/contributing/how-to-open-a-pull-request) before reading this section further.
+
+:::
+
+We have shared code of example app, that is run on different platforms and architectures. When working on an issue the most important destinations will be:
+
+- `apps/common-app` &ndash; shared source code with example app
+- `packages/react-native-reanimated/src/reanimated2` &ndash; containing source code related to JS part of Reanimated features
+- `packages/react-native-reanimated/android/src/main/java/com/swmansion/reanimated` &ndash; containing source code related to Android native part of Reanimated features
+- `packages/react-native-reanimated/apple` &ndash; containing source code related to iOS native part of Reanimated features
+
+Our workflow usually starts from editing `app/src/examples/EmptyExample.tsx` with testing example for our new feature or a bug. This is the part that we often attach to PR so other developers can test the same use-case. Next, we run the `Example` (or `FabricExample` depending on the architecture) app and start development.
+
+:::info
+
+If you want implement new feature or fix bug, but still feel uninformed, after reading through this section, or could use a guidance you can reach Reanimated Team on [discord](https://discord.com/channels/464786597288738816/1216675825584181330).
+
+:::
+
+### Working on Android
+
+To start with, let's install all dependencies:
+
+1. `yarn && yarn build`
+2. `cd apps/common-app && yarn && cd ..`
+3. `cd paper-example && yarn`
+4. `yarn start` &ndash; make sure to start metro bundler before building the app in Android Studio.
+
+and open `react-native-reanimated/apps/paper-example/android` with Android Studio.
+
+<img src={useBaseUrl('/img/contributing/android_studio.png')} />
+
+The native source code of Reanimated can be found in `react-native-reanimated` module, in `/packages/react-native-reanimated/android/src/main/java/com/swmansion/reanimated`. Making sure the Metro builder is running. You can now build React Native app or debug native code.
+
+Project structure overview:
+
+```
+├── AndroidUIScheduler.java
+├── CopiedEvent.java
+├── MapUtils.java
+├── NativeMethodsHelper.java
+├── NodesManager.java
+├── ReactNativeUtils.java
+├── ReanimatedMessageQueueThreadBase.java
+├── ReanimatedPackage.java
+├── ReanimatedUIManagerFactory.java
+├── Utils.java
+├── keyboard
+├── layoutReanimation
+├── nativeProxy
+└── sensor
+```
+
+### Working on iOS
+
+To begin with, let's install all the necessary dependencies:
+
+1. `yarn && yarn build`
+2. `cd apps/common-app && yarn && cd ..`
+3. `cd paper-example && yarn`
+4. `yarn start` &ndash; make sure to start metro bundler before building the app in Xcode.
```

Comment:
It's kinda wrong. Firstly, we don't need yarn in `common-app` or `paper-example`. Secondly, we need to install the pods.
```suggestion
1. `yarn && yarn build`
2. `cd paper-example/ios && bundle install && bundle exec pod install` &ndash; install Pods for iOS
3. `cd .. && yarn start` &ndash; make sure to start metro bundler before building the app in Xcode.
```